### PR TITLE
Do not update alert timestamp when updating a triggered alert

### PR DESCRIPTION
**Pull request by louis-oui**
DO NOT DELETE THIS TEXT

#### Please note

> Please read this information carefully. You can run `./scripts/pre-commit.php` to check your code before submitting.

- [X] Have you followed our [code guidelines?](http://docs.librenms.org/Developing/Code-Guidelines/)

#### Testers

If you would like to test this pull request then please run: `./scripts/github-apply <pr_id>`, i.e `./scripts/github-apply 5926`
After you are done testing, you can remove the changes with `./scripts/github-remove`.  If there are schema changes, you can ask on discord how to revert.

### CODE REVIEW

**Comment by louis-oui** [comment]
alerts table is configured to update timestamp whenever an entry is updated because of "on update CURRENT_TIMESTAMP" parameter

```
MariaDB [librenms]> desc alerts;
+-----------+------------------+------+-----+-------------------+-----------------------------+
| Field     | Type             | Null | Key | Default           | Extra                       |
+-----------+------------------+------+-----+-------------------+-----------------------------+
| id        | int(10) unsigned | NO   | PRI | NULL              | auto_increment              |
| device_id | int(10) unsigned | NO   | MUL | NULL              |                             |
| rule_id   | int(10) unsigned | NO   | MUL | NULL              |                             |
| state     | int(11)          | NO   |     | NULL              |                             |
| alerted   | int(11)          | NO   |     | NULL              |                             |
| open      | int(11)          | NO   |     | NULL              |                             |
| note      | text             | YES  |     | NULL              |                             |
| timestamp | timestamp        | NO   |     | CURRENT_TIMESTAMP | on update CURRENT_TIMESTAMP |
| info      | text             | NO   |     | NULL              |                             |
```

When polling change the state from 0 to 1 or 1 to 0, it is OK to change timestamp because timestamp reflects time the alert is triggered or recovered.

However, when alerts.php update alerts entries, we won't change to the change the timestamp as the alert is not new. Case alerts.php updates alerts entries are:
- state column is changed to value 3 or 4 : respectively "worse" and "better" state
- alert column is set to state value

**Comment by louis-oui** [comment]
Code climate failed because of legacy code or Laravel external code

File: database/migrations/2019_12_05_164700_alerts_disable_on_update_current_timestamp.php [hunk 1]
```diff
@@ -0,0 +1,28 @@
...
+{
+
+    /**
+     * Run the migrations.
+     *
+     * @return void
+     */
+    public function up()
+    {
+        \DB::statement("ALTER TABLE `alerts` CHANGE `timestamp` `timestamp` timestamp NOT NULL default CURRENT_TIMESTAMP;");
```
Comment (Jellyfrog) [inline database/migrations/2019_12_05_164700_alerts_disable_on_update_current_timestamp.php]: Please use the "real" migration way,
`$this->timestamp('timestamp')->useCurrent();` maybe works.

Comment (louis-oui) [reply database/migrations/2019_12_05_164700_alerts_disable_on_update_current_timestamp.php]: Thank you for the review
I got this issue https://stackoverflow.com/questions/34774628/how-do-i-make-doctrine-support-timestamp-columns
Tried 
```
        Schema::table('alerts', function (Blueprint $table) {
            $table->datetime('timestamp')->useCurrent()->change();
        });
```

Comment (Jellyfrog) [reply database/migrations/2019_12_05_164700_alerts_disable_on_update_current_timestamp.php]: I see. Annoying.

**Comment by louis-oui** [comment]
Before merging, we need to make sure that date in file is last one in list database/migrations/2019_12_05_164700_alerts_disable_on_update_current_timestamp.php
Otherwise, migration won't work

**Comment by louis-oui** [comment]
@Jellyfrog Please let me know what do you want to do with this PR ?

**Comment by louis-oui** [comment]
@Jellyfrog any news on it ?

**Comment by murrant** [comment]
This pull request has been mentioned on **LibreNMS Community**. There might be relevant details there:

https://community.librenms.org/t/do-not-change-alert-timestamp-when-acknowledge/10698/1

**Comment by murrant** [comment]
This pull request has been mentioned on **LibreNMS Community**. There might be relevant details there:

https://community.librenms.org/t/v1-60-release-changelog-january-2020/10863/1

**Pull request merged**